CMakeLists.txt

```diff
@@ -627,21 +627,14 @@ if(WITH_INFLATE_ALLOW_INVALID_DIST)
 endif()
 
 #
-# Macro to add either the given intrinsics option to the global compiler options,
+# Macro to add the given intrinsics option to the specified files,
 # or ${NATIVEFLAG} (-march=native) if that is appropriate and possible.
-# An alternative version of this macro would take a file argument, and set ${flag}
-# only for that file as opposed to ${NATIVEFLAG} globally, to limit side-effect of
-# using ${flag} globally.
 #
 macro(add_intrinsics_option flag)
     if(WITH_NATIVE_INSTRUCTIONS AND NATIVEFLAG)
-        if(NOT "${CMAKE_C_FLAGS} " MATCHES ".*${NATIVEFLAG} .*")
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${NATIVEFLAG}")
-        endif()
+        set_property(SOURCE ${ARGN} PROPERTY COMPILE_FLAGS ${NATIVEFLAG})
     else()
-        if(NOT "${CMAKE_C_FLAGS} " MATCHES ".*${flag} .*")
-            set(CMAKE_C_FLAGS "${CMAKE_C_FLAGS} ${flag}")
-        endif()
+        set_property(SOURCE ${ARGN} PROPERTY COMPILE_FLAGS ${flag})
```

Comment:
and here?

Reply:
As far as I understand, this is used to test functionality of single flag or group of flags, so any old value should be ignored.